### PR TITLE
fix builds on Ubuntu-24.04

### PR DESCRIPTION
**Pull request by lyskov**

### CODE REVIEW

**Review by roccomoretti — APPROVED** [review]
There might be better ways of fixing this, but it seems reasonable as it stands.

File: source/tools/build/basic.settings [hunk 1]
```diff
@@ -2198,6 +2198,17 @@ settings = {
         },
     },
 
+    "clang, |cxx_ver:>=18|" : {
+        "appends" : {
+            "flags" : {
+                "warn" : [
+                    "Wno-deprecated-declarations",
+		    "Wno-enum-constexpr-conversion",
```
[the remark this answers is not in the document]
Comment (lyskov) [reply source/tools/build/basic.settings:2206]: The issue is was with boost libraries. I have tried to upgrade related libraries but the only way to fix it was to upgrade to versions that require C++14. I added discussion topic for the next devel meeting to discuss this.

[the remark this answers is not in the document]
Comment (lyskov) [reply source/tools/build/basic.settings:2206]: Also, +1 on `Wno-error` idea, - i will try that!

**Comment by lyskov** [comment]
Thank you for review and suggestions @roccomoretti !

**Pull request merged**